Allow routes to accept routes as strings

Aethylred opened the pull request:
This change gives routes the same behaviour as rules where they can be a mix of strings and hashes and allows setting of rules and routes that don't fit the expected formats. Examples used in the documentation were to configure the rules for a HAproxy setup as per [Transparent Proxying and Binding with HAProxy](https://www.haproxy.com/blog/howto-transparent-proxying-and-binding-with-haproxy-and-aloha-load-balancer/)

Aethylred commented:
This was also easier to do that figure out how to get it to pass through `local`, `locahost`, `default`, and other strings that are valid for rules and routes. In it's current form the role only accepts IP addresses.

markgoddard commented on README.md:
````diff
@@ -445,6 +445,40 @@ bouncing network interfaces before they are active. This can be done by setting
+
+### host_vars/proxy2
+
+```yaml
````
Empty

markgoddard commented on README.md:
````diff
@@ -445,6 +445,40 @@ bouncing network interfaces before they are active. This can be done by setting
+```yaml
+interfaces_route_tables:
+    - name: frontend
+      id: 200
````
Should it be 100?

Aethylred replied:
oops, no there should be a 
```
    - name: marktable
      id: 100
```

I mixed two 'sources' and didn't standardise on named tables

markgoddard commented on templates/route_RedHat.j2:
```diff
@@ -18,5 +19,8 @@
 {%   set route = route ~ ' ' ~ option %}
 {% endfor %}
+{% else %}
+{% set route = i%}
```
nit
```suggestion
{% set route = i %}
```

markgoddard reviewed (changes requested):
Strictly we should have similar support for Debian, in `templates/*_Debian.j2`, and network manager, in `templates/*_nmconnection.j2`

Aethylred commented:
The example rules are not exactly perfect, they're roles for the `lo` device, but defined on the `eth1` device, but the module doesn't really have a method for setting rules independently of devices. Adding a `interfaces_extra_rules` was more work than I had time for

Aethylred commented:
I think I can follow up with the Debian and nmconnection templates, but it looks like they don't consistently have the string-or-mapping behaviour as described for rules, `bond_Debian.j2` is missing it for example.

Aethylred commented:
@markgoddard I've addressed the points you brought up, I've also got a few ideas that would improve consistency in the templates, but this will do for now.
